crates/volta-core/src/run_package_global/executor.rs

```diff
@@ -259,3 +260,45 @@ impl From<InternalInstallCommand> for Executor {
         Executor::InternalInstall(Box::new(cmd))
     }
 }
+
+/// Executor for running a tool uninstall command.
+///
+/// This will use the `volta uninstall` logic to correctly ensure that the package is fully
+/// uninstalled
+pub struct UninstallCommand {
+    tool: Spec,
+}
+
+impl UninstallCommand {
+    pub fn new(tool: Spec) -> Self {
+        UninstallCommand { tool }
+    }
+
+    /// Runs the uninstall with Volta's internal uninstall logic
+    fn execute(self) -> Fallible<ExitStatus> {
+        info!(
+            "{} using Volta to uninstall {}",
+            note_prefix(),
+            tool_name(&self.tool)
+        );
+
+        self.tool.uninstall()?;
+
+        Ok(ExitStatus::from_raw(0))
+    }
+}
+
+impl From<UninstallCommand> for Executor {
+    fn from(cmd: UninstallCommand) -> Self {
+        Executor::Uninstall(Box::new(cmd))
+    }
+}
+
+fn tool_name(tool: &Spec) -> &str {
+    match tool {
+        Spec::Node(_) => "Node",
+        Spec::Npm(_) => "npm",
+        Spec::Yarn(_) => "Yarn",
+        Spec::Package(name, _) => &name,
+    }
+}
```

Comment:
**Suggestion:** make this a `pub fn` on `Spec` so that we can just write `&self.tool.name()` above; it seems likely it's the kind of thing that may be useful in the future but in any case clarifies the relationship now.

Reply:
Done, good idea!